src/System.Linq.Parallel/tests/Helpers/IntegerRangeSet.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Xunit;
+
+namespace Test
+{
+    // Dummy psuedo-set for verifying we've seen all of a range of integers, and only once.
+    internal class IntegerRangeSet : IEnumerable<KeyValuePair<int, bool>>
+    {
+        private BitArray _seen;
+        private int _start;
+        private object _locker;
+
+        public IntegerRangeSet(int start, int count)
+        {
+            _start = start;
+            _seen = new BitArray(count);
+            _locker = new object();
+        }
+
+        public bool Add(int entry)
+        {
+            Assert.InRange(entry, _start, _start + _seen.Length - 1);
+
+            lock (_locker)
+            {
+                Assert.False(_seen[entry - _start]);
+                return _seen[entry - _start] = true;
```

Comment:
I use the Boolean in some of the tests, actually.  I use it in `Any(...)` and `All(...)` to verify that it sees all elements (and a few other places too); it's easier to just use the return value than to have to do a multi-statement lambda.